### PR TITLE
feat: add user context

### PR DESCRIPTION
**Pull request by thomaszurkan-optimizely**
Summary
-------

- Added user context and create off of optimizely.  Added unit tests as well.

Test plan
---------
Unit tests added

### CODE REVIEW

File: tests/test_user_context.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2019, Optimizely
```
Comment (aliabbasrizvi) [inline tests/test_user_context.py:1]: 2020

File: optimizely/user_context.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+#
```
Comment (aliabbasrizvi) [inline optimizely/user_context.py:1]: Omit this line.

File: optimizely/user_context.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        """
+        sets a attribute by key for this user context.
+        Args:
+          attribute_key: key to use for attribute
+          attribute_value: attribute value
+
+        Returns:
+        None
+        """
+        self.user_attributes[attribute_key] = attribute_value
```
Comment (jaeopt) [inline optimizely/user_context.py:48]: Can we add a mutex control for attributes read/write?  
"setAttributes" and attribute -reading for decide can happen in other threads. See C#-sdk as a good reference (https://github.com/optimizely/csharp-sdk/pull/253)

File: tests/test_user_context.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        """
+        tests user context creating and attributes
+        """
+        uc = UserContext(self.optimizely, "test_user")
+        self.assertEqual(uc.user_attributes, {}, "should have created default empty")
+        self.assertEqual(uc.user_id, "test_user", "should have same user id")
+        uc.set_attribute("key", "value")
+        self.assertEqual(uc.user_attributes["key"], "value", "should have added attribute")
+        uc.set_attribute("key", "value2")
+        self.assertEqual(uc.user_attributes["key"], "value2", "should have new attribute")
```
Comment (jaeopt) [inline tests/test_user_context.py:34]: Can we have more tests:
- construct with non-empty attributes
- setAttribute can override the init attributes
- attributes not changed once constructed when the caller copy is updated

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good. 
Should add mutex control for attributes. More tests suggested.

File: tests/test_user_context.py (hunk 4 above)
Comment (oakbani) [reply tests/test_user_context.py:34]: Done in 314

**Comment by jaeopt** [comment]
Merged with #309

**Pull request closed without merging**